corehq/apps/data_interfaces/tests/test_case_deduplication.py

```diff
@@ -0,0 +1,486 @@
+                ("Padme Amidala", "1901-05-01", "Ashla"),
+                ("Padme Amidala", "1901-05-01", "Light"),
+                ("Anakin Skywalker", "1977-03-25", "Dark"),
+                ("Darth Vadar", "1977-03-25", "Dark"),
+            ]
+        ]
+
+        self._prime_es_index(cases)
+
+        # Padme cases are of different case types
+        self.assertNotIn(
+            cases[1].case_id,
+            [case.case_id for case in find_duplicate_cases(self.domain, cases[0], ["name", "dob"])]
+        )
+
+        # Anakin / Vadar are still the same
+        self.assertIn(
+            cases[3].case_id, [case.case_id for case in find_duplicate_cases(self.domain, cases[2], ["dob"])]
+        )
+
+    def test_find_closed_duplicates(self):
+        """closed duplicates should or shouldn't be found based on input
+        """
+
+        cases = [
+            self.factory.create_case(case_name=case_name, close=closed, update={'dob': dob})
+            for (case_name, dob, closed) in [
+                ("Padme Amidala", "1901-05-01", False),
+                ("Padme Amidala", "1901-05-01", True),
+                ("Anakin Skywalker", "1977-03-25", False),
+                ("Darth Vadar", "1977-03-25", True),
+            ]
+        ]
+
+        self._prime_es_index(cases)
+
+        # Even though the Padme case is closed, it should still be flagged as a duplicate
+        self.assertItemsEqual([cases[0].case_id, cases[1].case_id], [
+            case.case_id
+            for case in find_duplicate_cases(self.domain, cases[0], ["name", "dob"], include_closed=True)
+        ])
+
+        # The Vadar case is closed, so shouldn't be returned
+        self.assertItemsEqual([cases[2].case_id], [
+            case.case_id for case in find_duplicate_cases(self.domain, cases[2], ["dob"], include_closed=False)
+        ])
+
+    def test_find_duplicates_any_rule(self):
+        """find duplicates where any case properties match
+        """
+        cases = [
+            self.factory.create_case(case_name=case_name, update={'dob': dob}) for (case_name, dob) in [
+                ("Padme Amidala", "1901-05-01"),
+                ("Padme Amidala", "1901-05-01"),
+                ("Padme Amidala", "1901-05-02"),
+                ("Padme Naberrie", "1901-05-01"),
+                ("Anakin Skywalker", "1977-03-25"),
+                ("Darth Vadar", "1977-03-25"),
+            ]
+        ]
+
+        self._prime_es_index(cases)
+
+        self.assertItemsEqual([cases[0].case_id, cases[1].case_id, cases[2].case_id, cases[3].case_id], [
+            case.case_id
+            for case in find_duplicate_cases(self.domain, cases[0], ["name", "dob"], match_type="ANY")
+        ])
+
+        self.assertItemsEqual([cases[4].case_id, cases[5].case_id], [
+            case.case_id
+            for case in find_duplicate_cases(self.domain, cases[4], ["name", "dob"], match_type="ANY")
+        ])
+
+
+@override_settings(TESTS_SHOULD_USE_SQL_BACKEND=True)
```

Comment:
this shouldn't be necessary anymore https://github.com/dimagi/commcare-hq/pull/30411

Reply:
👍🏽